pg_shard--1.2.sql

```diff
@@ -1,53 +1,140 @@
-/* pg_shard--1.1.sql */
+/* pg_shard--1.2.sql */
 
 -- complain if script is sourced in psql, rather than via CREATE EXTENSION
 \echo Use "CREATE EXTENSION pg_shard" to load this file. \quit
 
--- the pgs_distribution_metadata schema stores data distribution information
-CREATE SCHEMA pgs_distribution_metadata
-
-	-- shard keeps track of hash value ranges for each shard
-	CREATE TABLE shard (
-		id bigint primary key,
-		relation_id oid not null,
-		storage "char" not null,
-		min_value text not null,
-		max_value text not null
-	)
-
-	-- shard_placement records which nodes contain which shards
-	CREATE TABLE shard_placement (
-		id bigint primary key,
-		shard_id bigint not null references shard(id),
-		shard_state integer not null,
-		node_name text not null,
-		node_port integer not null
-	)
-
-	-- partition lists a partition key for each distributed table
-	CREATE TABLE partition (
-		relation_id oid unique not null,
-		partition_method "char" not null,
-		key text not null
-	)
-
-	-- make a few more indexes for fast access
-	CREATE INDEX shard_relation_index ON shard (relation_id)
-	CREATE INDEX shard_placement_node_name_node_port_index
-		ON shard_placement (node_name, node_port)
-	CREATE INDEX shard_placement_shard_index ON shard_placement (shard_id)
-
-	-- make sequences for shards and placements
-	CREATE SEQUENCE shard_id_sequence MINVALUE 10000 NO CYCLE
-	CREATE SEQUENCE shard_placement_id_sequence NO CYCLE;
-
--- mark each of the above as config tables to have pg_dump preserve them
-SELECT pg_catalog.pg_extension_config_dump(
-	'pgs_distribution_metadata.shard', '');
-SELECT pg_catalog.pg_extension_config_dump(
-	'pgs_distribution_metadata.shard_placement', '');
-SELECT pg_catalog.pg_extension_config_dump(
-	'pgs_distribution_metadata.partition', '');
+-- needed in our views
+CREATE FUNCTION column_to_column_name(table_oid oid, column_var text)
+RETURNS text
+AS 'MODULE_PATHNAME'
+LANGUAGE C STABLE STRICT;
+
+DO $$
+DECLARE
+    replication_factor integer;
+    use_citus_metadata boolean := false;
+BEGIN
+    BEGIN
+        replication_factor = current_setting('shard_replication_factor')::integer;
+        use_citus_metadata = (replication_factor IS NOT NULL);
+    EXCEPTION
+        WHEN undefined_object THEN
+            use_citus_metadata = false;
+    END;
+
+    IF use_citus_metadata THEN
```

Comment:
These functions mirror what we're doing in code as well. It seems wasteful to have similar logic replicated across both. If the SPI machinery would allow us to keep this logic only here (and keep the C metadata functions simpler), then I'd prefer that. Of course, that is predicated upon whether using SPI causes other locking issues (or is prohibitive due to other reasons).
It will not match Citus metadata functions, but I am OK with that. @ozgune Any concerns there?